lib/invisible_captcha/controller_ext.rb

```diff
@@ -22,6 +31,14 @@ def on_spam_action(options = {})
       end
     end
 
+    def on_threshold_spam_action(options = {})
+      if action = options[:on_threshold_spam]
+        send(action)
+      else
+        flash[:error] = InvisibleCaptcha.threshold_error_message
```

Comment:
Yes, but since we asume that submit is potentially spam, is it really necessary to keep the previous values?  

Also, this can be changed via callback:

```
on_threshold_spam: :my_callback     

def my_callback
  render "edit"
end
```

If we want to "automagically" discover/render the form that triggered the filter, we should pass over the session the action_name or something like that, not a really nice solution ... and I'm still not sure what's best as a default response...